File: app/src/main/java/space/pal/sig/old/view/splash/SplashActivity.java
package space.pal.sig.old.view.splash;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;

import androidx.lifecycle.ViewModelProvider;

import javax.inject.Inject;

import butterknife.BindView;
import space.pal.sig.R;
import space.pal.sig.old.Space;
import space.pal.sig.old.view.SpaceBaseActivity;
import space.pal.sig.old.view.main.MainActivity;

/**
 * SpaceNow
 * Created by Catalin on 7/22/2020
 **/
public class SplashActivity extends SpaceBaseActivity {

    @BindView(R.id.splash_progress_bar)
    ProgressBar progressBar;
    @Inject
    SplashViewModelFactory factory;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        Space.getApplicationComponent().inject(this);
        SplashViewModel splashViewModel = new ViewModelProvider(this, factory)
                .get(SplashViewModel.class);
        splashViewModel.isSafeForStart().observe(this, safeForStart -> {
            progressBar.setVisibility(safeForStart ? View.INVISIBLE : View.VISIBLE);
            if (safeForStart) {
                startApp();
            }
        });
    }

    private void startApp() {
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/space/pal/sig/view/splash/SplashActivity.kt
package space.pal.sig.view.splash

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import androidx.core.view.isVisible
import androidx.databinding.DataBindingUtil
import org.koin.android.ext.android.inject
import space.pal.sig.R
import space.pal.sig.databinding.ActivitySplashBinding
import space.pal.sig.view.BaseActivity
import space.pal.sig.view.main.MainActivity

class SplashActivity : BaseActivity() {

    companion object {
        const val SPLASH_SYNC_PROGRESS_ACTION = "SplashActivity.Broadcast.SyncProgress.Action"
        const val SYNC_MESSAGE = "SplashActivity.SyncMessage"
    }

    private lateinit var binding: ActivitySplashBinding
    private val viewModel: SplashViewModel by inject()

    private val splashBroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            intent?.let {
                val message = intent.getStringExtra(SYNC_MESSAGE)
                binding.splashSyncMessage.text = message
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this,
                R.layout.activity_splash)
        registerSplashReceiver()
        viewModel.isSafeForStart().observe(this) { openMainActivity(it) }
    }

    override fun onDestroy() {
        unregisterReceiver(splashBroadcastReceiver)
        super.onDestroy()
    }

    private fun registerSplashReceiver() {
        val intentFilter = IntentFilter(SPLASH_SYNC_PROGRESS_ACTION)
        registerReceiver(splashBroadcastReceiver, intentFilter)
    }

    private fun openMainActivity(canOpen: Boolean) {
        binding.splashProgressBar.isVisible = !canOpen
        if (canOpen) {
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }
}